Protectli vault ehl variant

Stojek139808 opened the pull request.

macpijan commented:
This covers base directory structure and GPIO configuration. The rest is pretty much WIP and copied from vault_glk?

macpijan commented:
@Stojek139808 We should not target `master` branch. You can create `develop` branch for EHL and set it as a base in this MR.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+
+	/* GPP_T8 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_T8, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_T9 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_T9, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_T10 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_T10, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_T11 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_T11, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_T12 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_T12, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_T13 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_T13, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_T14 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_T14, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_T15 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_T15, NONE, DEEP, TxLASTRxE, SAME),
```
All pads NC

Stojek139808 replied:
Done.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+
+	/* GPP_G7 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_G7, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_G8 - RSVD */
+	/* DW0: 0x44000b00, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_G8, NONE, DEEP, NF2),
+
+	/* GPP_G9 - RSVD */
+	/* DW0: 0x44000b00, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_G9, NONE, DEEP, NF2),
+
+	/* GPP_G10 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_G10, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_G11 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_G11, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_G12 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_G12, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_G13 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_G13, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_G14 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_G14, NONE, DEEP, TxLASTRxE, SAME),
```
All these pads are NC

Stojek139808 replied:
Done.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	/* GPP_G19 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_G19, NONE, DEEP, TxLASTRxE, SAME),
```
This is SPI TPM Interrupt so it should be:
```suggestion
	PAD_CFG_GPI_APIC_LOCK(GPP_G19, NONE, LEVEL, INVERT, LOCK_CONFIG),
```

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_G22, NONE, DEEP, NF1),
+
+	/* GPP_G23 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_G23, NONE, DEEP, TxLASTRxE, SAME),
```
PAD_NC

Stojek139808 replied:
Done.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_V11, UP_20K, DEEP, NF1),
+
+	/* GPP_V12 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_V12, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_V13 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_V13, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_V14 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_V14, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_V15 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_V15, NONE, DEEP, TxLASTRxE, SAME),
```
Pads are NC

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	PAD_CFG_GPIO_HI_Z(GPP_H1, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_H2 - GPIO */
+	/* DW0: 0x44000200, DW1: 0x00000000 */
+	PAD_CFG_GPO(GPP_H2, 0, DEEP),
+
+	/* GPP_H3 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H3, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_H4 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H4, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_H5 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H5, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_H6 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H6, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_H7 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H7, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_H8 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H8, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_H9 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H9, NONE, DEEP, TxLASTRxE, SAME),
```
Pads are NC

Stojek139808 replied:
Done.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H11, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_H12 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H12, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_H13 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H13, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_H14 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H14, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_H15 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H15, NONE, DEEP, TxLASTRxE, SAME),
```
Pads are NC

Stojek139808 replied:
Done.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) | (1 << 1) - IGNORED */
+	PAD_CFG_NF(GPP_H16, NONE, DEEP, NF2),
+
+	/* GPP_H17 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H17, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_H18 - PMC_CPU_C10_GATE_N */
+	/* DW0: 0x44000700, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_H18, NONE, DEEP, NF1),
```
Pads are NC

Stojek139808 replied:
Done.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) | (1 << 1) - IGNORED */
+	PAD_CFG_NF(GPP_H20, NONE, DEEP, NF2),
+
+	/* GPP_H21 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H21, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_H22 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H22, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_H23 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_H23, NONE, DEEP, TxLASTRxE, SAME),
```
Pads are NC

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+
+	/* GPP_D12 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_D12, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_D13 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_D13, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_D14 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_D14, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_D15 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_D15, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_D16 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_D16, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_D17 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_D17, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_D18 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_D18, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_D19 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_D19, NONE, DEEP, TxLASTRxE, SAME),
```
Pads are NC

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+
+	/* GPP_U11 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_U11, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_U12 - ISI_CHX_OKNOK_0 */
+	/* DW0: 0x44000700, DW1: 0x00001000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_U12, DN_20K, DEEP, NF1),
+
+	/* GPP_U13 - ISI_CHX_OKNOK_1 */
+	/* DW0: 0x44000700, DW1: 0x00001000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_U13, DN_20K, DEEP, NF1),
+
+	/* GPP_U14 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_U14, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_U15 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_U15, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_U16 - ISI_OKNOK_0 */
+	/* DW0: 0x44600702, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) | (1 << 1) - IGNORED */
+	PAD_CFG_NF(GPP_U16, NONE, DEEP, NF1),
+
+	/* GPP_U17 - ISI_OKNOK_1 */
+	/* DW0: 0x44600702, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) | (1 << 1) - IGNORED */
+	PAD_CFG_NF(GPP_U17, NONE, DEEP, NF1),
+
+	/* GPP_U18 - ISI_ALERT_N */
+	/* DW0: 0x44600702, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) | (1 << 1) - IGNORED */
+	PAD_CFG_NF(GPP_U18, NONE, DEEP, NF1),
```
Pads are NC, BTW GPP_U19 is missing.

Stojek139808 replied:
Good you noticed that U19 is missing. In inteltool it had `GP_U19` set as a name, then intelp2m didn't match that name. Fixed that in inteltool upstream.

Stojek139808 replied:
Done, also added that U19 with PAD_NC.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+
+	/* GPP_A16 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_A16, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_A17 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_A17, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_A18 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_A18, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_A19 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_A19, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_A20 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_A20, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_A21 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_A21, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_A22 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_A22, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_A23 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_A23, NONE, DEEP, TxLASTRxE, SAME),
```
All pads are NC

Stojek139808 replied:
Done.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+
+	/* GPP_C16 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_C16, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_C17 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_C17, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_C18 - SML_DATA0 */
+	/* DW0: 0x04000f02, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) | (1 << 1) - IGNORED */
+	PAD_CFG_NF(GPP_C18, NONE, PWROK, NF3),
+
+	/* GPP_C19 - SML_CLK0 */
+	/* DW0: 0x04000f02, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) | (1 << 1) - IGNORED */
+	PAD_CFG_NF(GPP_C19, NONE, PWROK, NF3),
+
+	/* GPP_C20 - SIO_UART2_RXD */
+	/* DW0: 0x44001302, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) | (1 << 1) - IGNORED */
+	PAD_CFG_NF(GPP_C20, NONE, DEEP, NF4),
+
+	/* GPP_C21 - SIO_UART2_TXD */
+	/* DW0: 0x44001300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_C21, NONE, DEEP, NF4),
+
+	/* GPP_C22 - SIO_UART2_RTS_N */
+	/* DW0: 0x44001300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_C22, NONE, DEEP, NF4),
+
+	/* GPP_C23 - SIO_UART2_CTS_N */
+	/* DW0: 0x44001302, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) | (1 << 1) - IGNORED */
+	PAD_CFG_NF(GPP_C23, NONE, DEEP, NF4),
```
Pads are NC

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	PAD_CFG_NF(GPP_F0, NONE, DEEP, NF1),
+
+	/* GPP_F1 - RSVD */
+	/* DW0: 0x44000700, DW1: 0x00003000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_F1, UP_20K, DEEP, NF1),
+
+	/* GPP_F2 - RSVD */
+	/* DW0: 0x44000700, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_F2, NONE, DEEP, NF1),
+
+	/* GPP_F3 - RSVD */
+	/* DW0: 0x44000700, DW1: 0x00003000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_F3, UP_20K, DEEP, NF1),
+
+	/* GPP_F4 - RSVD */
+	/* DW0: 0x44000700, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_F4, NONE, DEEP, NF1),
+
+	/* GPP_F5 - RSVD */
+	/* DW0: 0x44000700, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_F5, NONE, DEEP, NF1),
+
+	/* GPP_F6 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_F6, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_F7 - GPIO */
+	/* DW0: 0x44000200, DW1: 0x00000000 */
+	PAD_CFG_GPO(GPP_F7, 0, DEEP),
+
+	/* GPP_F8 - ISI_TRACEDATA_0 */
+	/* DW0: 0x44001700, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_F8, NONE, DEEP, NF5),
```
Pads are NC

Stojek139808 replied:
Done.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+
+	/* GPP_F16 - ISI_SWCLK */
+	/* DW0: 0x44001702, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) | (1 << 1) - IGNORED */
+	PAD_CFG_NF(GPP_F16, NONE, DEEP, NF5),
+
+	/* GPP_F17 - ISI_TRACEDATA_3 */
+	/* DW0: 0x44001700, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_F17, NONE, DEEP, NF5),
+
+	/* GPP_F18 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_F18, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_F19 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_F19, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_F20 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_F20, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_F21 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_F21, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_F22 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_F22, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_F23 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_F23, NONE, DEEP, TxLASTRxE, SAME),
```
Pads are NC

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_E0, NONE, PLTRST, NF1),
+
+	/* GPP_E1 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_E1, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_E2 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_E2, NONE, DEEP, TxLASTRxE, SAME),
```
Pads are NC

Stojek139808 replied:
Done.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_E3, NONE, DEEP, NF1),
+
+	/* GPP_E4 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_E4, NONE, DEEP, TxLASTRxE, SAME),
```
Pad NC

Stojek139808 replied:
Done.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) | (1 << 1) - IGNORED */
+	PAD_CFG_NF(GPP_E5, NONE, DEEP, NF1),
+
+	/* GPP_E6 - GPIO */
+	/* DW0: 0x44000200, DW1: 0x00000000 */
+	PAD_CFG_GPO(GPP_E6, 0, DEEP),
```
Pad NC

Stojek139808 replied:
Done.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_E7, NONE, DEEP, NF1),
+
+	/* GPP_E8 - SATA_1_DEVSLP */
+	/* DW0: 0x04000b00, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_E8, NONE, PWROK, NF2),
+
+	/* GPP_E9 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_E9, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_E10 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_E10, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_E11 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_E11, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_E12 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_E12, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_E13 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_E13, NONE, DEEP, TxLASTRxE, SAME),
```
Pads are NC

Stojek139808 replied:
Done.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) | (1 << 1) - IGNORED */
+	PAD_CFG_NF(GPP_E14, NONE, DEEP, NF1),
+
+	/* GPP_E15 - RSVD */
+	/* DW0: 0x44000b00, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_E15, NONE, DEEP, NF2),
+
+	/* GPP_E16 - RSVD */
+	/* DW0: 0x44000b00, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_E16, NONE, DEEP, NF2),
+
+	/* GPP_E17 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_E17, NONE, DEEP, TxLASTRxE, SAME),
```
Pads are NC

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_E19, NATIVE, DEEP, NF1),
+
+	/* GPP_E20 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_E20, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_E21 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_E21, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_E22 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_E22, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_E23 - GPIO */
+	/* DW0: 0x44000200, DW1: 0x00000000 */
+	PAD_CFG_GPO(GPP_E23, 0, DEEP),
```
Pads are NC

Stojek139808 replied:
Done.

miczyg1 commented on src/mainboard/protectli/vault_ehl/gpio.h:
```diff
@@ -0,0 +1,1664 @@
+	/* DW0: PAD_TRIG(OFF) | PAD_BUF(TX_RX_DISABLE) - IGNORED */
+	PAD_CFG_NF(GPP_R4, NONE, DEEP, NF1),
+
+	/* GPP_R5 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_R5, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_R6 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_R6, NONE, DEEP, TxLASTRxE, SAME),
+
+	/* GPP_R7 - GPIO */
+	/* DW0: 0x44000300, DW1: 0x00000000 */
+	/* DW0: PAD_TRIG(OFF) - IGNORED */
+	PAD_CFG_GPIO_HI_Z(GPP_R7, NONE, DEEP, TxLASTRxE, SAME),
```
Pads are NC

Stojek139808 replied:
Done.

Stojek139808 commented:
> @Stojek139808 We should not target `master` branch. You can create `develop` branch for EHL and set it as a base in this MR.

Sorry, that was an oversight.

> This covers base directory structure and GPIO configuration. The rest is pretty much WIP and copied from vault_glk?

Yes, like I was instructed I based this port on glk, on first commit I've copied files from it. Then I've adjusted configs, commented out conflicting code (now removed) and changed what I could while doing that so it compiles. Later also added GPIO configuration.

Stojek139808 commented on src/mainboard/protectli/vault_ehl/Kconfig:
```diff
@@ -0,0 +1,50 @@
+config MAINBOARD_PART_NUMBER
+	string
+	default "VP2420"
+
```
Added.

Stojek139808 commented on src/mainboard/protectli/vault_ehl/Kconfig:
```diff
@@ -0,0 +1,50 @@
+
+
+config MAINBOARD_DIR
+	string
```
Fixed.

Stojek139808 commented on src/mainboard/protectli/vault_ehl/Kconfig:
```diff
@@ -0,0 +1,50 @@
+
+config DEVICETREE
+	string
+	default "devicetree.cb"
```
Removed.

Stojek139808 commented on configs/config.protectli_vp2420:
```diff
@@ -0,0 +1,22 @@
+CONFIG_BOARD_PROTECTLI_VP2420=y
+CONFIG_UART_PCI_ADDR=0x0
+CONFIG_CPU_MICROCODE_CBFS_EXTERNAL_BINS=y
+CONFIG_CPU_UCODE_BINARIES="3rdparty/intel-microcode/intel-ucode/06-96-01"
```
Updated, now it generates ucode.

Stojek139808 commented on src/mainboard/protectli/vault_ehl/Kconfig:
```diff
@@ -0,0 +1,50 @@
+config PXE_ROM_ID
+	string
+	default "8086,157b"
+
```
Removed.

Stojek139808 commented on src/mainboard/protectli/vault_ehl/bootblock.c:
```diff
@@ -0,0 +1,22 @@
+{	
+	ite_reg_write(GPIO_DEV, 0x2c, 0x41); /* disable k8 power seq */
+	ite_reg_write(GPIO_DEV, 0x2d, 0x02); /* PCICLK 25MHz */
+	//ite_enable_serial(UART_DEV, CONFIG_TTYS0_BASE);
```
Forgot to uncomment it as it generated an error before. Now it's uncommented and builds successfully.

Stojek139808 commented on src/mainboard/protectli/vault_ehl/board.fmd:
```diff
@@ -0,0 +1,22 @@
+FLASH 8M {
```
Yes I am aware, but since I don't have the information required to change the rest of this file right now, I've just left it as is for now.

miczyg1 replied:
Then remove it completely as it will not work as intended on this platform.

miczyg1 commented:
@Stojek139808 just remove baord.fmd file and remove it from Kconfig and we can merge
